### DISCUSSION TITLE
How to render underline on space letters?

### DISCUSSION THREAD

**Discussion by Reverier-Xu** [sourcelair/xterm.js#4462] [Q&A] [answered]
Hi, thanks for your awesome work!

I'm using xterm.js 5.1.0 with plugins and options:

```typescript
const term = new Terminal({} as ITerminalOptions)

const fitAddon = new FitAddon()
const weblinksAddon = new WebLinksAddon()
const canvasAddon = new CanvasAddon()

term.loadAddon(fitAddon)
term.loadAddon(weblinksAddon)
term.loadAddon(canvasAddon)
```

I found that xterm.js cannot draw underline on blank(space) characters:

![image](https://user-images.githubusercontent.com/41937333/229163586-9d6f4b60-bfd4-4c48-922a-971a9c2db10b.png)

I could not find any option to change this behavior, but I found that vscode can display the underline properly. What should I do?

**Comment by Tyriar** [comment]
This actually works fine when you use a space:

<img width="237" alt="image" src="https://user-images.githubusercontent.com/2193314/229206801-5bdcc7cc-39dc-4f32-92b3-2159c0a78e06.png">

I suspect the gap is a tab in your image which is why it's not working?

**Reply by Reverier-Xu** [reply]
![image](https://user-images.githubusercontent.com/41937333/229279016-43d7edd4-8479-47dc-995d-bf4222446fab.png)

nether `tab` nor `space` works :(

**Reply by Reverier-Xu** [reply]
I wrote a minimal demo to reproduce here, I could confirm that it does not work in all my browsers:

https://github.com/Reverier-Xu/xtermjs-4462

Microsoft Edge 111.0.1661.62
![image](https://user-images.githubusercontent.com/41937333/229279756-251cd1fe-f151-4ef0-b0cd-d737f6718bdd.png)

Firefox 111.0.1
![image](https://user-images.githubusercontent.com/41937333/229279787-9d9ff037-314f-4618-be37-e31679719acf.png)

Chromium 111.0.5563.146
![image](https://user-images.githubusercontent.com/41937333/229279804-bf808691-49a3-4f5b-8e45-7b4f5d2eba13.png)

**Reply by Tyriar** [reply]
Could you try on the latest beta tag?

**Reply by jerch** [reply, marked as answer]
This is prolly related to https://github.com/xtermjs/xterm.js/pull/4438  :grimacing:

**Reply by Reverier-Xu** [reply]
![image](https://user-images.githubusercontent.com/41937333/229527102-99da91d6-b699-46f3-b936-53418d31f75d.png)


~~The lastest beta tag doesn't seem to work either...~~

I upgrade `xterm-addon-canvas` to `0.4.0-beta.9`, it works now!
